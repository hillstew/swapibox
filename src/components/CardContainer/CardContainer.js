import React from "react";
import { Card } from "../Card/Card";

export const CardContainer = props => {
  let cards = props[props.view].map(subject => {
    return <Card subject={subject} view={props.view} />;
  });

  return <div>{cards}</div>;
};
