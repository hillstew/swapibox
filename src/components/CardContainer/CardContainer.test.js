import React from "react";
import CardContainer from "../CardContainer/CardContainer";
import { shallow } from "enzyme";

const mockPeople = [];
const mockPlanets = [];
const mockVehicles = [
  {
    class: "wheeled",
    model: "Digger Crawler",
    name: "Sand Crawler",
    passengers: "30"
  },
  {
    class: "repulsorcraft",
    model: "T-16 skyhopper",
    name: "T-16 skyhopper",
    passengers: "1"
  }
];
const mockView = "vehicles";

describe("CardContainer", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <CardContainer
        planets={mockPlanets}
        view={mockView}
        vehicles={mockVehicles}
        people={mockPeople}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
