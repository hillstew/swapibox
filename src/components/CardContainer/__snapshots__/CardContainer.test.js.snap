// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
<div>
  <Card
    key="Sand Crawler"
    subject={
      Object {
        "class": "wheeled",
        "model": "Digger Crawler",
        "name": "Sand Crawler",
        "passengers": "30",
      }
    }
    view="vehicles"
  />
  <Card
    key="T-16 skyhopper"
    subject={
      Object {
        "class": "repulsorcraft",
        "model": "T-16 skyhopper",
        "name": "T-16 skyhopper",
        "passengers": "1",
      }
    }
    view="vehicles"
  />
</div>
`;
